atc/exec/put_inputs.go

```diff
@@ -51,40 +51,36 @@ func NewSpecificInputs(inputs []string) PutInputs {
 	}
 }
 
-func (i specificInputs) FindAll(artifacts *artifact.Repository) ([]worker.InputSource, error) {
+func (i specificInputs) FindAll(artifacts *build.Repository) ([]worker.FooBarInput, error) {
 	artifactsMap := artifacts.AsMap()
 
-	inputs := []worker.InputSource{}
+	inputs := []worker.FooBarInput{}
 	for _, i := range i.inputs {
-		artifactSource, found := artifactsMap[artifact.Name(i)]
+		artifact, found := artifactsMap[build.ArtifactName(i)]
 		if !found {
 			return nil, PutInputNotFoundError{Input: i}
 		}
 
-		inputs = append(inputs, &putInputSource{
-			name:   artifact.Name(i),
-			source: PutResourceSource{artifactSource},
+		inputs = append(inputs, &putFooBarInput{
+			name:     build.ArtifactName(i),
+			artifact: artifact,
 		})
 	}
 
 	return inputs, nil
 }
 
-type putInputSource struct {
-	name   artifact.Name
-	source worker.ArtifactSource
+type putFooBarInput struct {
+	name     build.ArtifactName
+	artifact runtime.Artifact
 }
 
-func (s *putInputSource) Source() worker.ArtifactSource { return s.source }
+func (s putFooBarInput) Artifact() runtime.Artifact { return s.artifact }
 
-func (s *putInputSource) DestinationPath() string {
+func (s putFooBarInput) DestinationPath() string {
 	return resource.ResourcesDir("put/" + string(s.name))
 }
 
-type PutResourceSource struct {
-	worker.ArtifactSource
-}
-
-func (source PutResourceSource) StreamTo(ctx context.Context, logger lager.Logger, dest worker.ArtifactDestination) error {
-	return source.ArtifactSource.StreamTo(ctx, logger, dest)
-}
+//func (source PutResourceSource) StreamTo(ctx context.Context, logger lager.Logger, dest worker.ArtifactDestination) error {
+//	return source.StreamableArtifactSource.StreamTo(ctx, logger, dest)
+//}
```

Comment:
For funzies I decided to see if I could figure out what a `FooBarInput` was just from reading this code. It sounds like it's either the full set, or a subset, of registered artifacts that are currently registered in the ArtifactRepository object, which may be used as part of the output generation process for exporting some artifact to the outside world during a Put. If that's true, then maybe the concept here is like a `StepProcessingInputs`, or `StepContainerInputs`, or `ActiveInputs`.

Not sure if you have historical context on this but I'm wondering why there's added complexity to cut down the set of inputs, especially when we're prepared to default it to the whole set when it's not specified. Is it related to performance?

On the whole I think it was a good idea to pull this logic out into its own file and introduce some interfaces. The same can probably be done for task step inputs later!

Reply:
me and sameer had an impromptu discussion about this.

Heres the other options we considered:
```
DesiredInputs
InputRequests
ConfiguredInputs
InputClaims
InputRef
InputReferences
ArtifactReferences
```

We decided to change the implementation where FooBarInputs would be a map[DestinationPath]runtime.Artifact instead of a []FooBarInputs{artifact, destinationPath}

But we still didn't decide a name for this map.